src/ansys/sound/core/sound_composer/source_spectrum.py

```diff
@@ -0,0 +1,245 @@
+            str_source = (
+                f"'{self.source_spectrum_data.name}'\n"
+                f"\tFmax: {support_data[-1]:.0f} Hz\n"
+                f"\tDeltaF: {support_data[1] - support_data[0]:.1f} Hz"
+            )
+        else:
+            str_source = "Not set"
+
+        if self.is_source_control_valid():
+            str_source_control = (
+                f"{self.source_control.get_method_name()}, {self.source_control.duration} s"
+            )
+        else:
+            str_source_control = "Not set/valid"
+
+        return f"Spectrum source: {str_source}\nSource control: {str_source_control}"
+
+    @property
+    def source_control(self) -> SourceControlSpectrum:
+        """Spectrum source control object.
+
+        Contains the duration in seconds, and the method used to generate the sound.
+        """
+        return self.__source_control
+
+    @source_control.setter
+    def source_control(self, source_control: SourceControlSpectrum):
+        """Set the spectrum source control."""
+        if not (isinstance(source_control, SourceControlSpectrum) or source_control is None):
+            raise PyAnsysSoundException(
+                "Specified source control object must be of type ``SourceControlSpectrum``."
+            )
+        self.__source_control = source_control
+
+    @property
+    def source_spectrum_data(self) -> Field:
+        """Spectrum source data, as a DPF field.
+
+        Power spectral density (PSD) as a DPF field, which contains the frequencies in Hz and
+        the levels in unit^2/Hz (for example Pa^2/Hz).
+        """
+        return self.__source_spectrum_data
+
+    @source_spectrum_data.setter
+    def source_spectrum_data(self, source_spectrum_data: Field):
+        """Set the spectrum source data, from a DPF field."""
+        if not (isinstance(source_spectrum_data, Field) or source_spectrum_data is None):
+            raise PyAnsysSoundException(
+                "Specified spectrum source must be provided as a DPF field."
+            )
+        self.__source_spectrum_data = source_spectrum_data
+
+    def is_source_control_valid(self) -> bool:
+        """Source control verification function.
+
+        Check if the source control is set and its duration is strictly positive.
+
+        Returns
+        -------
+        bool
+            True if the source control is set and its duration is strictly positive, False
+            otherwise.
+        """
+        return (self.source_control is not None) and (self.source_control.duration > 0.0)
+
+    def load_source_spectrum(self, file_source: str):
+        """Load the spectrum source data from a spectrum file.
+
+        Parameters
+        ----------
+        file_source : str
+            Path to the spectrum file. Supported files are XML files and text files with the header
+            AnsysSound_Spectrum, as supported by Ansys Sound SAS.
```

Comment:
@anssmolla you changed this part, but I feel the phrasing is ambiguous on two points:
- does the "with the header AnsysSound_Spectrum" part only refer to text files, or both XML and text files?
- "as supported by SAS" do you mean to say the same file types that are supported in SAS? or something else?

Reply:
@ansaminard 1. the header is only for txt files, 2. correct, what I mean is that we support here the same txt and xml files as the ones supported in SAS
Maybe @mattbennett72 can tell us how to better formulate it in English, if current text is not clear enough?

Reply:
@ansaminard @anssmolla how about:

Supported files are the same XML and text (with the AnsysSound_Spectrum header) formats as supported by Ansys Sound SAS.

Reply:
@mattbennett72 @anssmolla Perfect, thanks! Let me change that accordingly